akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/MiscDirectivesSpec.scala

```diff
@@ -65,6 +65,72 @@ class MiscDirectivesSpec extends RoutingSpec {
     }
   }
 
+  "the withSizeLimit directive" should {
+    "not apply if entity is not consumed" in {
+      val route = withSizeLimit(500) { completeOk }
+
+      Post("/abc", entityOfSize(500)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", entityOfSize(501)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+    }
+
+    "apply if entity is consumed" in {
+      val route = withSizeLimit(500) {
+        entity(as[String]) { _ ⇒
+          completeOk
+        }
+      }
+
+      Post("/abc", entityOfSize(500)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", entityOfSize(501)) ~> Route.seal(route) ~> check {
+        status shouldEqual StatusCodes.BadRequest
+      }
+    }
+
+    "properly handle nested directives" in {
+      val route =
+        withSizeLimit(500) {
+          withSizeLimit(800) {
+            entity(as[String]) { _ ⇒
+              completeOk
+            }
+          }
+        }
+
+      Post("/abc", entityOfSize(800)) ~> route ~> check {
+        status shouldEqual StatusCodes.OK
+      }
+
+      Post("/abc", entityOfSize(801)) ~> Route.seal(route) ~> check {
+        status shouldEqual StatusCodes.BadRequest
+      }
+    }
+  }
+
+  "the withoutSizeLimit directive" should {
```

Comment:
OK, will try to rewrite test I included in my comment in `ClientServerSpec` manner